### PR TITLE
Fix arguments object detection in non-complex param list

### PR DESCRIPTION
**Pull request by rerobika**
This patch is the followup of #4849.

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik robert.fancsik@h-lab.eu

### CODE REVIEW

File: tests/jerry/es.next/arguments.js [hunk 1]
```diff
@@ -217,3 +217,8 @@ function f22 (arguments, [a = arguments]) {
   assert(arguments === 3.1);
 }
 f22(3.1, []);
+
+function f23(arguments, eval = () => eval()) {
+  assert(arguments === 3.1);
```
Comment (zherczeg) [inline tests/jerry/es.next/arguments.js:222]: I would prefer an undefined comparison to prove that the arguments object has never been created (and not overwritten later by the number argument)

**Review by zherczeg — CHANGES_REQUESTED** [review]
This fix is better.

File: tests/jerry/es.next/arguments.js (hunk 1 above)
Comment (rerobika) [reply tests/jerry/es.next/arguments.js]: updated.

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by ossy-szeged — APPROVED** [review]

**Pull request merged**